Gems/Atom/RHI/Code/Source/RHI/Image.cpp

```diff
@@ -107,6 +107,23 @@ namespace AZ::RHI
     const RHI::Ptr<RHI::DeviceImageView> ImageView::GetDeviceImageView(int deviceIndex) const
     {
         AZStd::lock_guard lock(m_imageViewMutex);
+
+        if (m_image->GetDeviceMask() != m_deviceMask)
+        {
+            m_deviceMask = m_image->GetDeviceMask();
+
+            for (auto checkDeviceIndex{ 0 }; checkDeviceIndex < RHISystemInterface::Get()->GetDeviceCount(); ++checkDeviceIndex)
+            {
+                if (!m_image->IsDeviceSet(checkDeviceIndex))
+                {
+                    if (auto it{ m_cache.find(checkDeviceIndex) }; it != m_cache.end())
+                    {
+                        m_cache.erase(it);
+                    }
+                }
+            }
+        }
```

Comment:
Can we refactor this code into a function as this seems to be similar to Buffer's implementation